File: sort/bubble.py
def bubble(a):
    for i in range(len(a) - 1):
        for j in range(len(a) - 1):
            if a[j] > a[j + 1]:
                temp = a[j]
                a[j] = a[j + 1]
                a[j + 1] = temp
    return a


#冒泡排序测试
nums = [32, 31, 29, 26, 20, 24, 35, 6, 25, 39]
print("sort before:")
print(nums)
nums = bubble(nums)
print("sort after:")
print(nums)

File: sort/bubble.cpp
#include <iostream>
using namespace std;

void bubble(int a[], int n)
{
    for (int i = 0; i < n - 1; i++)
    {
        for (int j = 0; j < n - 1; j++)
        {
            if (a[j] > a[j + 1])
            {
                swap(a[j], a[j + 1]);
            }
        }
    }
}

void output(int a[], int n)
{
    for (int i = 0; i < 10; i++)
    {
        cout << a[i] << " ";
    }
    cout << endl;
}

int main()
{
    int a[] = {32, 31, 29, 26, 20, 24, 35, 6, 25, 39};
    cout << "sort before:" << endl;
    output(a, 10);
    bubble(a, 10);
    cout << "sort after:" << endl;
    output(a, 10);
    return 0;
}